### PR TITLE
Reduce noise in PR template

### PR DESCRIPTION
**Pull request by brisvag**
# Description

Trying to reduce noise in the PR template both for contributors and for reviewers.

My reasoning is:
- the fix/feature/maintenance checkboxes are adding nothing to the description, and reviewers need to add the equivalent labels anyways now.
- the "how has this been tested" section is also never used, and redundant given that CI takes care of testing and saying whether tests pass or not. If new tests need to be added, the checklist comment is enough as a reminder, unless the tests required are too complex, in which case a reviewer needs to help out anyways.
- having the checklist as actual visible checklist is pretty useless: no one ever completes it or updates it after submission, and if anything is missing it's always the reviewers pointing it out anyways.
- the Fixes/Closes section was recently moved to the top because of how hard it was to find due to all the noise. I find that now it's fine to fit it below the description together with References.

This template will remove all that boilerplate (that everyone ignores anyways) and make PRs more clear and concise.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,39 +1,20 @@
...
 # Description
-<!-- What does this pull request (PR) do? Why is it necessary? -->
-<!-- Tell us about your new feature, improvement, or fix! -->
-<!-- If your change includes user interface changes, please add an image, or an animation "An image is worth a thousand words!" -->
-<!-- You can use https://www.cockos.com/licecap/ or similar to create animations -->
-
-# References
-<!-- What resources, documentation, and guides were used in the creation of this PR? -->
+<!-- What does this pull request (PR) do? Is it a new feature, a bug fix,
+an improvement, or someting else? Why is it necessary? If relevant, please add
```
Comment (lucyleeow) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
an improvement, or something else? Why is it necessary? If relevant, please add
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,39 +1,20 @@
...
+a screenshot or a screen capture: "An image is worth a thousand words!" -->
```
Comment (lucyleeow) [inline .github/PULL_REQUEST_TEMPLATE.md]: If we are trying to be concise, should "An image is worth a thousand words" stay?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,39 +1,20 @@
...
-- [ ] I have commented my code, particularly in hard-to-understand areas
-- [ ] I have made corresponding changes to the documentation
-- [ ] I have added tests that prove my fix is effective or that my feature works
-- [ ] If I included new strings, I have used `trans.` to make them localizable.
-      For more information see our [translations guide](https://napari.org/developers/translations.html).
+<!-- Final Checklist
+- My PR is the minimum possible work for the desired functionality
+- I have commented my code, particularly in hard-to-understand areas
+- I have made corresponding changes to docstrings and documentation
+  (open a PR on the napari/docs repository if changes are necessary there!)
```
Comment (lucyleeow) [inline .github/PULL_REQUEST_TEMPLATE.md]: Should we link to 'napari/docs' repo?

**Comment by Czaki** [comment]
> the Fixes/Closes section was recently moved to the top because of how hard it was to find due to all the noise. I find that now it's fine to fit it below the description together with References.

It is the same place where it was before and @jni told that it is hard to find it here when using mobile. 



> the fix/feature/maintenance checkboxes are adding nothing to the description, and reviewers need to add the equivalent labels anyways now.

my plan is to automatically add labels based on this. And remove non checked. 



> having the checklist as actual visible checklist is pretty useless: no one ever completes it or updates it after submission, and if anything is missing it's always the reviewers pointing it out anyways.

we may think about refactoring it and Denny review if not all points are satisfied.

**Review by lucyleeow — COMMENTED** [review]
I agree with your sentiments and this is definitely an improvement. I don't think I have enough napari background to comment on some parts.

I would say that the final checklist is a bit wordy (potentially it could be a quick checklist, e.g., 'adds tests', that directs you to our docs that expands? but also fine to stay as is). I also don't think it is really compatible if you're just making a quick typo/typing/doc change. I never know what to do in this checklist if my PR is one of the above.

**Comment by brisvag** [comment]
> It is the same place where it was before and @jni told that it is hard to find it here when using mobile.

I don't feel strongly about this, but I think the issue was also that it was *hard to find* because of how much noise there is. Like this, it's always at the bottom, so just scroll down :shrug:
 
> my plan is to automatically add labels based on this. And remove non checked.

We can re-add it together with that implementation once it's ready :)

> we may think about refactoring it and Denny review if not all points are satisfied.

This sounds basically impossibl (how are you going to check automatically any of those bullet points? Or do you mean manually? In which case we would have to click it ourselves? I don't see the usefulness... And still, even if that turns out to be useful and possible, we can always re-add the list once we *actually* have an implementation.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (brisvag) [reply .github/PULL_REQUEST_TEMPLATE.md]: Fair enough :P We could put just that instead of the verbose explanation in fact... I just didn't want to squeeze *too much* and come off as dry, but we can definitely reduce this still.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (brisvag) [reply .github/PULL_REQUEST_TEMPLATE.md]: Yes, good point.

**Comment by brisvag** [comment]
> I would say that the final checklist is a bit wordy (potentially it could be a quick checklist, e.g., 'adds tests', that directs you to our docs that expands?

I was trying to be concise but not too dry, but I don't mind either way.

> I don't think it is really compatible if you're just making a quick typo/typing/doc change. I never know what to do in this checklist if my PR is one of the above.

Yeah, one of the reasons why the "checkable" checklist feels wrong. This is more of a reminder of things you might have forgotten.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (lucyleeow) [reply .github/PULL_REQUEST_TEMPLATE.md]: Yes I understand, I don't have a strong feeling either way so I leave to you.

**Comment by lucyleeow** [comment]
> Yeah, one of the reasons why the "checkable" checklist feels wrong. This is more of a reminder of things you might have forgotten.

e.g., this very PR does not fit the checklist. Also maintenance work like CI or refactoring do not fit.

**Comment by lucyleeow** [comment]
>  Otherwise, we could meet a person claiming that he does not need to implement things pointed out by the community manager.

Interesting, I would have thought that the community manager would have more authority and flexibility than having a strict checklist?

**Comment by lucyleeow** [comment]
That sounds unfortunate, maybe even falls under 'code of conduct' ?

**Comment by brisvag** [comment]
> In general, I agree. But I already meet a person writing "I do not see such requirements in other PR so why do you require additional job from my side. You should accept it as it is". In general, it was not a nice discussion.

Having the checklist won't solve this, unless you enforce somehow that every pr matches every single point in the checklist... Which is never going to work (you don't *always* have to add tests, and you don't *always* have to add X amount of comments). Let's not get stuck in formalities more than necessary: we're not making the reviewing process easier but harder by adding such checks, only to avoid the occasional personal decision from reviewers, which should be totally fine.

**Review by GenevieveBuckley — APPROVED** [review]
Looks much cleaner, I'm on board.

**Review by Czaki — COMMENTED** [review]
I have opened #6157 that have a conflict with this PR, but I think that it is valuable to leave this section in the template.

**Comment by brisvag** [comment]
I find that noisy and redundant (since anyways the label with be there immediately after), and does not really make the work of reviewing any easier or faster. I'll leave the decisions to others, but at worst we can also reintroduce it later with the other PR.

**Comment by jni** [comment]
> Anyways only a core-dev can merge, which is the only moment when this is relevant no?

That's my view also, but @Carreau made the point in #6042 that it's frustrating as a contributor to not have the ability to get a PR to green status without help from the core devs. Personally I'm ok with it but I understand the concern.

**Review by jni — APPROVED** [review]

**Comment by brisvag** [comment]
Yeah, I see, that's a valid concern... In that case I would veer in favour of *not* having a check on PR but rather a check on release or somehow on the project tracker.

**Comment by Czaki** [comment]
> Yeah, I see, that's a valid concern... In that case I would veer in favour of _not_ having a check on PR but rather a check on release or somehow on the project tracker.

[NOOO](http://www.nooo.me/) Do not do this to the release manager.

**Comment by Carreau** [comment]
It's also frustrating as a maintainer, I tend to look at the list of issues, and prioritize looking first as the one with test failing and try to fix it (even if not my pr). And most of the PRs are red because of label, or yellow because of milestone, so I end up opening a dozen PR every morning, just to see they are actually all right.

I think we can find a good balance like 
 - "need label" label (sic) and
 - "need milestone" label 

We can also have a pinned issue that list all PRs that needs update.

As a reminder there is this PR (#5995) that start to automatically deal with labels, and is  waiting .

**Comment by Carreau** [comment]
Ha ! I think I have an idea. 

Is it possible to post a failing status for label/milestone ONLY if the label "ready to merge is present" ? 

This way: 
 - The user can get the PR all green, 
 - It can get reviewed. 
 - A maintainer apply "Ready to merge"
 - It run "maintainer checks" (labels, html stripping...), it could even already rebase squash...
 - And then it gets merged ?

This also let us be more liberal with whom we give "triage rights", as triage right is enough to label I think.

**Comment by brisvag** [comment]
Love this, let's do it!

**Comment by jni** [comment]
@Carreau, genius! 😍

**Comment by Czaki** [comment]
#6158 implements such check. The main weakness is that it is implemented using GitHub actions. so it may stuck in queue. But is it will provide better feeling then it may be worth to re implement it using GitHub App (that runs on separate machine).

**Comment by brisvag** [comment]
Assuming a version of #6158 goes in (since we all seem to agree), I think this leaves this PR as good to go in like this?

**Comment by Czaki** [comment]
I have changed milestone as PR template depends on main branch
